docs/_guide/events.md

```diff
@@ -8,96 +8,156 @@ slug: events
+}
+disconnectedCallback() {
+  window.removeEventListener('resize', this.handleResize);
+  super.disconnectedCallback();
+}
+```
+
+### Add event listeners after first paint
+
+Sometimes, you may want to defer adding an event listener until after first paint—for example, if you're adding a lot of listeners and first paint performance is critical.
+
+LitElement doesn't have a specific lifecycle callback called after first paint, but you can use this pattern with the `firstUpdated` lifecycle callback:
+
+```js
+async firstUpdated() {
+  // Give the browser a chance to paint
+  await new Promise((r) => setTimeout(r, 0));
+  this.addEventListener('click', this._handleClick);
+}
+```
+
+`firstUpdated` fires after the first time your component has been updated and called its `render` method, but **before the browser has had a chance to paint**. The `Promise`/`setTimeout` line yields to the browser
     
-    ```js
-    connectedCallback() {
-      super.connectedCallback();
-      document.addEventListener('readystatechange', this.handleChange);
-    }
-    disconnectedCallback() {
-      document.removeEventListener('readystatechange', this.handleChange);
-      super.disconnectedCallback();
-    }
-    ```
+See [firstUpdated](/guide/lifecycle#firstupdated) in the Lifecycle documentation for more information.
+
 
-### Using `this` in event handlers
+## Using `this` in event listeners
 
-The default JavaScript context object (`this`) inside an event handler belonging to a LitElement-based component is the component itself. 
+Event listeners added using the declarative (`@event`) syntax in the template are automatically _bound_ to the component.
 
-Therefore, you can use `this` to refer to your element instance inside any event handler:
+Therefore, you can use `this` to refer to your component instance inside any declarative event handler:
 
 ```js
 class MyElement extends LitElement {
   render() {
-    return html`<button @click="${this.handleClick}">click</button>`;
+    return html`<button @click="${this._handleClick}">click</button>`;
   }
-  handleClick(e) {
+  _handleClick(e) {
     console.log(this.prop);
   }
 }
 ```
 
+When adding listeners imperatively with `addEventListener`, you'll need to bind the event listener yourself if you need a reference to the component instance. For example:
+
+```js
+this.boundResizeHandler = this.handleResize.bind(this);
+window.addEventListener('resize', this.boundResizeHandler);
+```
+
 See the [documentation for `this` on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this) for more information.
 
-### Use cases
+## Setting event listener options
+
+When you add an event listener imperatively, using `addEventListener`, you can specify various event listener options. For example, to use a capture phase event listener in plain JavaScript you'd do something like this:
+
+```js
+someElement.addEventListener('click', this._handleClick, {capture: true});
```

Comment:
I would suggest using a `'scroll'` event and the `passive: true` option just 'cause more people should know about it (it eliminates jank from scroll listeners).

Reply:
My understanding was that `scroll` itself can't be canceled, so it doesn't need `passive`? I think you'd use it on `touchstart`, `touchend`, etc... https://developers.google.com/web/updates/2016/06/passive-event-listeners

Reply:
Used "touchstart" with passive.